src/main/java/nl/workingtalent/wtacademy/bookcopy/BookCopyController.java

```diff
@@ -20,38 +25,52 @@ public class BookCopyController {
 	@Autowired
 	private BookService bookService;
 	
-	@RequestMapping("book-copy/all")
-	public List<BookCopy> getAllBooks(){
-		
-		return service.getAllBookCopies();
+	@RequestMapping("bookcopy/all")
+	public Stream<ReadBookCopyDto> getAllBooks(){
+		List<BookCopy> copies = service.getAllBookCopies();
+ 		Stream<ReadBookCopyDto> dtos = copies.stream().map((copy)->{
+ 			return new ReadBookCopyDto(copy);
+ 		});
+		return dtos;
+
 	}
 	
-	@RequestMapping("book/book-copy/{bookId}")
-	public List<BookCopy> getAllCopiesForBookId(@PathVariable("bookId") long bookId){
-		
-		return service.getAllCopiesForBookId(bookId);
+	@RequestMapping("bookcopy/all/{bookId}")
+	public Stream<ReadBookCopyDto> getAllCopiesForBookId(@PathVariable("bookId") long bookId){
+ 		List<BookCopy> copies = service.getAllCopiesForBookId(bookId);
+ 		Stream<ReadBookCopyDto> dtos = copies.stream().map((copy)->{
+ 			return new ReadBookCopyDto(copy);
+ 		});
+		return dtos;
 	}
 	
-	@RequestMapping(method = RequestMethod.POST, value = "book-copy/create/{bookId}")
-	public void addBookCopy(@RequestBody BookCopy copy, @PathVariable("bookId") long bookId) {
-		if(bookService.getBookById(bookId).isEmpty()) return;
-		copy.setBook(bookService.getBookById(bookId).get());
+	@PostMapping("bookcopy/create")
+	public void addBookCopy(@RequestBody CreateBookCopyDto dto) {
+		Optional<Book> book = bookService.getBookById(dto.getBookId());
+		if(book.isEmpty()) return;
+		BookCopy copy = new BookCopy();
+		
+		copy.setState(dto.getState());
+		copy.setLocation(dto.getLocation());
+		copy.setBook(book.get());
+		
 		service.addBookCopy(copy);
 	}
 	
-	@RequestMapping(method = RequestMethod.PUT, value = "book-copy/update/{bookCopyId}")
-	public boolean updateBookCopy(@RequestBody BookCopy copy, @PathVariable("bookCopyId") long bookCopyId) {
-		if(service.getBookCopyById(bookCopyId).isEmpty()) return false;
-		BookCopy dbCopy = service.getBookCopyById(bookCopyId).get();
+	@PutMapping("bookcopy/update")
+	public boolean updateBookCopy(@RequestBody UpdateBookCopyDto dto) {
+		Optional<BookCopy> bookCopy = service.getBookCopyById(dto.getId());
+		if(bookCopy.isEmpty()) return false;
+		BookCopy dbCopy = service.getBookCopyById(dto.getId()).get();
 		
-		dbCopy.setCondition(copy.getCondition());
-		dbCopy.setLocation(copy.getLocation());
+		dbCopy.setState(dto.getState());
+		dbCopy.setLocation(dto.getLocation());
 		
 		service.addBookCopy(dbCopy);
 		return true;
 	}
 	
-	@RequestMapping(method = RequestMethod.DELETE, value = "book-copy/delete/{bookCopyId}")
+	@DeleteMapping("bookcopy/delete/{bookCopyId}")
 	public void deleteBookById(@PathVariable("bookCopyId") int bookCopyId) {
 		service.deleteBookCopyById(bookCopyId);
 	}
```

Comment:
Zouden er misschien responseDtos gebruikt moeten worden die je returnt ipv de dtos zelf?

Reply:
Dat is hoe wij het gedaan hebben in user

Reply:
Toegevoegd!